db-service/lib/cqn2sql.js

```diff
@@ -719,14 +734,36 @@ class CQN2SQLRenderer {
    */
   UPSERT(q) {
     const { UPSERT } = q
+    const entity = this.name(q.target?.name || UPSERT.into.ref[0])
```

Comment:
Why `q.target?.name` ?

Reply:
My question would rather be the other way around. Why it still has the fallback to the `into` as now `infer` should always put the `target` onto the query with the correct name even when the `target` is flagged as unresolved. As `q.target` should always be properly inferred allow `UPSERT` to be called on path expressions.

```suggestion
    const entity = this.name(q.target.name)
```